tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/environment/AbstractDockerBasedEnvironment.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized.environment;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import org.hawkular.apm.tests.dockerized.exception.EnvironmentException;
+import org.hawkular.apm.tests.dockerized.model.TestEnvironment;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateNetworkResponse;
+import com.github.dockerjava.api.exception.DockerException;
+import com.github.dockerjava.api.model.Network;
+import com.github.dockerjava.core.DockerClientBuilder;
+
+/**
+ * @author Pavol Loffay
+ */
+public abstract class AbstractDockerBasedEnvironment implements TestEnvironmentExecutor {
+    private static final Logger log = Logger.getLogger(AbstractDockerBasedEnvironment.class.getName());
+
+    protected String scenarioDirectory;
+    protected String apmBindAddress;
+    protected Network network;
+
+    /**
+     * Docker config
+     */
+    protected final DockerClient dockerClient;
+
+
+    public AbstractDockerBasedEnvironment(String scenarioDirectory, String apmBindAddress) {
+        this.scenarioDirectory = scenarioDirectory;
+        this.apmBindAddress = apmBindAddress;
+        this.dockerClient = DockerClientBuilder.getInstance().build();
+    }
+
+
+    @Override
+    public void close() {
+        if (network != null) {
+            try {
+                log.info(String.format("Removing network: %s", network.getName()));
+                dockerClient.removeNetworkCmd(network.getId()).exec();
```

Comment:
Actually there should not be a chance tahat `dockerClient` is null. I will remove the null check below.